fast/stages/1-resman/stage-2-networking.tf

```diff
@@ -60,35 +60,29 @@ module "net-folder" {
     # network security stage 2 service accounts
     var.fast_stage_2.network_security.enabled != true ? {} : {
       "roles/serviceusage.serviceUsageAdmin" = [
-        try(module.nsec-sa-rw[0].iam_email, null)
+        module.nsec-sa-rw[0].iam_email
```

Comment:
nsec is network security, which can be enabled/disabled independently from security; the trys are needed here...

Reply:
I know they're different things, and I think the roles/service accounts are mixed up. Looking at the security stage, I don't see anything related to firewall policies. `nsec`, on the other hand, does create firewall policies.

I don't see why the `try`s are needed. The SAs are supposed to exist if their stage is enabled, and the conditional around them guarantees the stage is enabled. Am I missing something?

In any case, at the very least, all these `try`s need a `compact` around the outer list; otherwise, you might end up with null elements, which makes IAM throw errors.

Reply:
Well, look at the current master. I just ported what is there.

Reply:
In master it's also only granted when nsec is enabled:

https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/6b0745deed68b63ca0d69c891cb4141768a94a7e/fast/stages/1-resman/branch-networking.tf#L33-L48

Reply:
Try serves same purpose, on mobile will send proper comments tmw 😂

Reply:
no hurry, I'm just curious...

Reply:
ok I misread the PR (mobile, teaches me do detach when I'm not at my coputer)...

- the condition is correct and trys not needed
- the firewall policy admin role *is* needed as in Luca's design (which we will change eventually) the nsec stage manages fw policies and needs to attach them to the VPCs defined here
- the nsec ro SA was missing a role, and I'm pretty sure I added it but you're right, it's not in bootstrap so either we a) add it, or b) use a looser but still view only predefined role

Reply:
> the condition is correct and trys not needed

Awesome :)

> the firewall policy admin role is needed as in Luca's design (which we will change eventually) the nsec stage manages fw policies and needs to attach them to the VPCs defined here

Correct, and it's already there.

> the nsec ro SA was missing a role, and I'm pretty sure I added it but you're right, it's not in bootstrap so either we a) add it, or b) use a looser but still view only predefined role

Also correct. My impression is that Luca replaced it with a `roles/compute.orgFirewallPolicyUse`, which is also already granted.